WordPress/Classes/Services/PeopleService.swift

```diff
@@ -156,24 +180,19 @@ struct PeopleService {
     ///     - failure: Closure to be executed on error
     ///
     func deleteUser(_ user: User, success: (() -> Void)? = nil, failure: ((Error) -> Void)? = nil) {
```

Comment:
These optional parameters required the new `delete(_ user:, using...` method added in this PR to also have optional `success` and `failure`. 

I did a shallow Xcode search which showed this method has only one caller:

<img width="407" alt="image" src="https://user-images.githubusercontent.com/1218433/213363807-8d6f8401-07d6-40a6-a47a-80424d4c962d.png">

That caller passes actual, non-optional values: 
https://github.com/wordpress-mobile/WordPress-iOS/blob/230140d8449c981859235a6a7c1f2cb926e4d7f6/WordPress/Classes/ViewRelated/People/PersonViewController.swift#L276-L284

So, I think it would be in-scope for this PR (or a quick followup) to remove the optionality, which I think would improve the code because it doesn't seem wise to allow people to fire off a request to delete a resource without waiting to know how it went. 

Of course, one can always implement an empty callback, but that shows next-level intention for ignoring things. Requiring a callback seems more in line with our aspiration to _make it easy to do the right thing_.

Reply:
Good catch. I've changed a couple places in https://github.com/wordpress-mobile/WordPress-iOS/pull/19947/commits/20bb63fdcadaec0bf408258a3ef3859ccb72996b